packages/shared/src/components/modals/useBootPopups.tsx

```diff
@@ -0,0 +1,179 @@
+import { useContext, useEffect, useState } from 'react';
+import { useLazyModal } from '../../hooks/useLazyModal';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { useActions, useBoot } from '../../hooks';
+import { ActionType } from '../../graphql/actions';
+import { LazyModal } from './common/types';
+import AlertContext from '../../contexts/AlertContext';
+import { MarketingCtaVariant } from '../cards/MarketingCta/common';
+import { AnalyticsEvent, TargetType } from '../../lib/analytics';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import { promotion } from './generic';
+import { useReadingStreak, useStreakExperiment } from '../../hooks/streaks';
+import { useSettingsContext } from '../../contexts/SettingsContext';
+
+const REP_TRESHOLD = 250;
+
+/**
+ * Component that renders popups that should be shown on boot.
+ * Set rule is that we only show one popup per day, and they come in first come, first served manner.
+ * These popups are removed when the users interact with page (clicks outside, closes, completes action).
+ * @constructor
+ */
+export const useBootPopups = (): void => {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const { checkHasCompleted, isActionsFetched, completeAction } = useActions();
+  const { openModal } = useLazyModal();
+  const { user } = useAuthContext();
+  const [shownModal, setShownModal] = useState(false);
+  const { alerts, loadedAlerts, updateAlerts } = useContext(AlertContext);
+  const [bootPopups, setBootPopups] = useState(new Map());
+  const { getMarketingCta } = useBoot();
+  const marketingCta = getMarketingCta(MarketingCtaVariant.Popover);
+  const { shouldShowPopup: shouldShowStreaksPopup } = useStreakExperiment();
+  const { streak, isEnabled } = useReadingStreak();
+  const { loadedSettings, optOutWeeklyGoal } = useSettingsContext();
+
+  // hide modal if settings are not loaded
+  let shouldHideStreaksModal = !loadedSettings;
+  // hide modal if feature is not enabled or user opted out
+  shouldHideStreaksModal =
+    shouldHideStreaksModal || !isEnabled || optOutWeeklyGoal;
+  // hide modal if user already closed it
+  shouldHideStreaksModal =
+    shouldHideStreaksModal || alerts?.showStreakMilestone !== true;
+  // hide modal if there's no streak
+  shouldHideStreaksModal = shouldHideStreaksModal || !streak?.current;
```

Comment:
I prefer better variable naming rather then comments as they can easily become outdated. I think naming only few of them differently and leaving the other ones as they are seems enough for the conditions to be fully understood.

```suggestion
  const streakModalClosed = alerts?.showStreakMilestone !== true
  const noCurrentStreak = !streak?.current

  let shouldHideStreaksModal = !loadedSettings || !isEnabled || optOutWeeklyGoal || streakModalClosed || noCurrentStreak;
```

Reply:
This was already made like this from the previous thing/
If I remember someone back then requested @kopancek to do it like this.
And want to avoid changing too much logic in favor of ensuring same evaluations as much as possible.